dask/array/slicing.py

```diff
@@ -209,7 +209,16 @@ def slice_with_newaxes(out_name, in_name, blockdims, index):
         for k, v in dsk.items():
             if k[0] == out_name:
                 k2 = (out_name,) + expand(k[1:], 0)
-                dsk2[k2] = Task(k2, v.func, v.args[0], expand_orig(v.args[1], None))
+                if isinstance(v.args[1], Alias):
+                    # positional indexing with newaxis
+                    indexer = expand_orig(dsk[v.args[1].key].value[1], None)
+                    tok = "shuffle-taker-" + tokenize(indexer)
+                    dsk2[tok] = DataNode(tok, (1, indexer))
```

Comment:
Do we reuse this across multiple tasks?

Reply:
Yes, all the tasks that used the old shuffle-taker thing will use the new shuffle-taker thing instead